presto-main/src/main/java/com/facebook/presto/util/Reflection.java

```diff
@@ -113,4 +127,22 @@ public static MethodHandle constructorMethodHandle(StandardErrorCode errorCode,
             throw new PrestoException(errorCode, e);
         }
     }
+
+    /**
+     * Returns a MethodHandle corresponding to the specified constructor.
+     * <p>
+     * Warning: The way Oracle JVM implements producing MethodHandle for a constructor involves
+     * creating JNI global weak references. G1 processes such references serially. As a result,
+     * calling this method in a tight loop can create significant GC pressure and significantly
+     * increase application pause time.
+     */
+    public static MethodHandle constructorMethodHandle(StandardErrorCode errorCode, Constructor constructor)
```

Comment:
It's hard to list all of them, but the code path that triggers this issue goes through the `MethodHandleNatives.init()` call IIRC, so you can track from there.